lib/render/vp2ShaderFragments/shaderFragments.cpp

```diff
@@ -110,6 +110,11 @@ namespace
 // Fragment registration
 MStatus HdVP2ShaderFragments::registerFragments()
 {
+    if (MGlobal::mayaState() != MGlobal::kInteractive) {
+        // Only register fragments in interactive Maya sessions.
+        return MS::kSuccess;
+    }
+
     MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer();
```

Comment:
As I commented in the conversation, we should not reject registration/deregistration. For headless config, we should not really ask Viewport 2.0 to initialize. Therefore, the line can be changed to:

// We don't want to force Viewport 2.0 to be initialized.
MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer(false);

Reply:
With this change, `theRenderer->getFragmentManager()` and `theRenderer->getShaderManager()` will return `nullptr`. We should not treat them as failure as they are expected for headless config. So, we should adjust the code structure a bit and only do the work when the fragment manager and the shader manager (i.e. move the major body of function inside a `if (fragmentManager)` clause.